### DISCUSSION TITLE
Feasibility of Integrating Spring into Pulsar Sink Connector

### DISCUSSION THREAD

**Discussion by NurramoX** [apache/pulsar#22052] [Q&A]
Hello Pulsar Community,

I’m exploring the idea of integrating Spring functionalities, specifically for data mapping and persistence, directly into a Pulsar sink connector. The goal is to leverage Spring’s capabilities to streamline the process of transforming message data into DTOs and subsequently persisting them in a database.

Therefore I wanted to gather insights from the community:

- Has anyone attempted something similar or sees potential challenges with this approach?
- Are there alternative suggestions you’d recommend for achieving similar outcomes?

Your feedback would be invaluable in determining whether this path is worth pursuing and how best to approach it if so.

Thanks for sharing your thoughts and experiences.

**Comment by lhotari** [comment]
@onobc any comments on integrating Spring into a Pulsar Sink Connector?
